stdlib/Random/src/RNGs.jl

```diff
@@ -346,48 +339,32 @@ Return the default global random number generator (RNG).
 @inline default_rng() = TaskLocalRNG()
 @inline default_rng(tid::Int) = TaskLocalRNG()
 
-copy!(dst::Xoshiro, ::_GLOBAL_RNG) = copy!(dst, default_rng())
-copy!(::_GLOBAL_RNG, src::Xoshiro) = copy!(default_rng(), src)
-copy(::_GLOBAL_RNG) = copy(default_rng())
-
+# defined only for backward compatibility with pre-v1.3 code when `default_rng()` didn't exist;
+# `GLOBAL_RNG` was never really documented, but was appearing in the docstring of `rand`
+const GLOBAL_RNG = default_rng()
+
+# In v1.0, the GLOBAL_RNG was storing the seed which was used to initialize it; this seed was used to implement
+# the following feature of `@testset`:
+# > Before the execution of the body of a `@testset`, there is an implicit
+# > call to `Random.seed!(seed)` where `seed` is the current seed of the global RNG.
+# But the global RNG is now TaskLocalRNG() and doesn't store its seed; in order to not break `@testset`, we now
+# store the seed used in a call like `seed!(seed)` *without* an explicit RNG in `GLOBAL_SEED`; the wording of the
+# feature above was sufficiently unprecise (e.g. what exactly is the "global RNG"?) that this solution seems fine;
+# in a breaking julia version, we would probably remove GLOBAL_SEED and this `@testset` feature
```

Comment:
Really? I think this feature is really useful. I actually was just recently looking into ways to add more things to be reset at the start and end of each `@testset` (think additional RNGs and other global state). Unfortunately the only way I could see was to create my own `AbstractTestSet` subtype and then change every single `@testset` invocation to explicitly specify that type, which is rather a burden :-(

Reply:
Oh that's the first time i see positive feedback on this feature, it has usually been "meh" ! So this comment about removing this was more of a reflexion of what I thought was the general sentiment; and also the fact that since we don't use `MersenneTwister` as the global RNG, there is a mismatch between what we say:
>  Before the execution of the body of a @testset, there is an implicit call to Random.seed!(seed) where seed is the current seed of the global RNG.

and what we can do, because the "global RNG" doesn't have a seed (there are multiple global RNGs, one per task, and we currently store only one seed. 

But anyway, i'm very glad to see someone else finding this feature useful :) I will remove this comment,  and i have a plan to make the feature more robust.